core-java-modules/core-java-io-2/src/test/java/com/baeldung/readlargefile/ReadLargeFileUnitTest.java

```diff
@@ -93,6 +100,62 @@ public final void givenUsingApacheIo_whenStreamingThroughAFile_thenCorrect() thr
         logMemory();
     }
 
+    @Test
+    public void givenUsingBufferedReader_whenIteratingAFile_thenCorrect() throws IOException {
+        String fileName = "src/test/resources/myFile";
+
+        logMemory();
+        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
+            while (br.readLine() != null) {
+                // do something with each line
+            }
+        }
+        logMemory();
+    }
+
+    @Test
+    public void givenUsingNewBufferedReader_whenIteratingAFile_thenCorrect() throws IOException {
+        String fileName = "src/test/resources/myFile";
+
+        logMemory();
+        try (BufferedReader br = java.nio.file.Files.newBufferedReader(Paths.get(fileName))) {
+            while (br.readLine() != null) {
+                // do something with each line
+            }
+        }
+        logMemory();
+    }
+
+    @Test
+    public void givenUsingSeekableByteChannel_whenIteratingAFile_thenCorrect() throws IOException {
+        String fileName = "src/test/resources/myFile";
+
+        logMemory();
+        try (SeekableByteChannel ch = java.nio.file.Files.newByteChannel(Paths.get(fileName), StandardOpenOption.READ)) {
+            ByteBuffer bf = ByteBuffer.allocate(1000);
+            int i;
+            while ((i = ch.read(bf)) > 0) {
```

Comment:
We can simplify here:
`while (ch.read(bf) > 0)`

Reply:
Done